test/Microsoft.DocAsCode.Build.ManagedReference.WithPlugins.Tests/SplitClassPageToMemberLevelTest.cs

```diff
@@ -0,0 +1,159 @@
+                Assert.NotNull(model);
+
+                Assert.Equal("Hello world!", model.Metadata["meta"]);
+                Assert.Equal(3, model.Children.Count);
+            }
+            {
+                var outputRawModelPath = GetRawModelFilePath("CatLibrary.Cat`2.#ctor.yml");
+                Assert.True(File.Exists(outputRawModelPath));
+                var model = JsonUtility.Deserialize<ApiBuildOutput>(outputRawModelPath);
+                Assert.NotNull(model);
+
+                Assert.Equal("Hello world!", model.Metadata["meta"]);
+                Assert.Equal(MemberType.Constructor, model.Type);
+                Assert.Equal(3, model.Children.Count);
+            }
+        }
+
+        [Fact]
+        public void ProcessMrefWithTocShouldSucceed()
+        {
+            var files = new FileCollection(Directory.GetCurrentDirectory());
+            files.Add(DocumentType.Article, new[] { "TestData/mref/Namespace1.Class1`2.yml" }, "TestData/");
+            files.Add(DocumentType.Article, new[] { "TestData/mref/toc.yml" }, "TestData/");
+            BuildDocument(files);
+            {
+                var outputRawModelPath = GetRawModelFilePath("Namespace1.Class1`2.yml");
+                Assert.True(File.Exists(outputRawModelPath));
+                var model = JsonUtility.Deserialize<ApiBuildOutput>(outputRawModelPath);
+                Assert.NotNull(model);
+
+                Assert.Equal("Hello world!", model.Metadata["meta"]);
+                Assert.Equal(3, model.Children.Count);
+            }
+            {
+                var outputRawModelPath = GetRawModelFilePath("CatLibrary.Cat`2.#ctor.yml");
+                Assert.True(File.Exists(outputRawModelPath));
+                var model = JsonUtility.Deserialize<ApiBuildOutput>(outputRawModelPath);
+                Assert.NotNull(model);
+
+                Assert.Equal("Hello world!", model.Metadata["meta"]);
+                Assert.Equal(MemberType.Constructor, model.Type);
+                Assert.Equal(3, model.Children.Count);
+                Assert.Equal(new List<string> { "net2", "net46" }, model.Platform);
+            }
+            {
+                var outputRawModelPath = GetRawModelFilePath("toc.yml");
+                Assert.True(File.Exists(outputRawModelPath));
+                var model = JsonUtility.Deserialize<TocItemViewModel>(outputRawModelPath);
+                Assert.NotNull(model);
+                Assert.Equal(1, model.Items.Count);
+                Assert.Equal("Namespace1.Class1%602.html", model.Items[0].TopicHref);
+                Assert.Equal(1, model.Items[0].Items.Count);
+                Assert.Equal("CatLibrary.Cat%602.%23ctor.html", model.Items[0].Items[0].TopicHref);
+                Assert.Equal("Cat", model.Items[0].Items[0].Name);
+                Assert.Equal(new List<string> { "net2", "net46" }, JArray.FromObject(model.Items[0].Items[0].Metadata[Constants.PropertyName.Platform]).Select(s => s.ToString()).ToList());
```

Comment:
CatLibrary.Cat`2.#ctor(`0) has platform "net2"

---
In reply to: [102863453](https://github.com/dotnet/docfx/pull/1341#discussion_r102863453) [](ancestors = 102863453)